### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by peter-budo**
@karllindmark please merge as soon as possible

Extracted fragment/dialog fragment casting logic into BaseFragment

### CODE REVIEW

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/base/ui/BaseFragment.java [hunk 1]
```diff
@@ -79,6 +83,23 @@ protected Bundle getArgumentsBundle() {
         return getArguments() == null ? new Bundle() : getArguments();
     }
 
+    protected void castDetailFragment(final Bundle dataToPass, final FragmentFactory.Type fragmentType, final String tag) {
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/base/ui/BaseFragment.java]: I would like to suggest an alternative method signature:

`openDetailFragment(final FragmentFactory.Type fragmentType, final Bundle dataToPass, final String tag)`

Moving the `FragmentFactory.Type` argument to the left due to the fact that it's most important. :-)

**Comment by karllindmark** [comment]
@peter-budo: Could you add this behavior into the `BaseListFragment` as well? That way we'll get it to the stats as well. :-)

**Comment by peter-budo** [comment]
@karllindmark updated

**Comment by karllindmark** [comment]
Looks great, haven't been able to try it but I trust it works as it did before. :-)

**Pull request merged**